frontend/src/metabase/dashboard/components/DashboardInfoSidebar.tsx

```diff
@@ -67,12 +71,12 @@ const DashboardInfoSidebar = ({
     [saveDashboardAndCards, setDashboardAttribute],
   );
 
+  const dispatch = useDispatch();
   const handleToggleAutoApplyFilters = useCallback(
     (isAutoApplyingFilters: boolean) => {
-      setDashboardAttribute("auto_apply_filters", isAutoApplyingFilters);
-      saveDashboardAndCards(true);
+      dispatch(toggleAutoApplyFilters(isAutoApplyingFilters));
```

Comment:
Uses the same action when turning off auto-apply filters via the auto-apply filter toast.
https://github.com/metabase/metabase/blob/ed3c3baf639a6982cda1ad10456377ceaefe1370/frontend/src/metabase/dashboard/actions/parameters.js#L352